test/core/rules/test_cases/L036.yml

```diff
@@ -8,6 +8,17 @@ test_single_select_target_and_new_line_after_select_target:
     select *
     from x
 
+test_single_select_target_and_new_line_before_select_target:
+  fail_str: |
+      select
+          a
+      from x
+  # In a normal "sqlfluff fix" run, the extra spaces in fix_str would be removed
```

Comment:
In recent rule work, this has been coming up frequently, where a single rule by itself corrects something about the SQL, but leaves behind formatting issues that would normally be addressed by other rules, but are not addressed in these tests, which only run **one** rule. Two questions:
* Is it reasonable that a rule implicitly expects other rules to "tidy up" its fixes?
  * Pros: Allows individual rules to be simpler and more focused
  * Cons: Non-obvious "dependencies" between rules could be confusing for users.
* In these cases, is this a reasonable approach to testing, i.e. running **just** the one rule and documenting the ugly formatting? Another possibility would be to let a test for rule X also run other rules (perhaps just a specific set of rules designated using config).